### DISCUSSION TITLE
Use socketio with uWebsockets

### DISCUSSION THREAD

**Discussion by steveyout** [adonisjs/core#3639] [Help] [answered]
According to socketio you can use other websocket implementations.I want to use uWebsockets since i have seen its incredibly fast and performant.So how can i implement it in adonisjs applications.I followed the socketio cookbook in the adonisjs docs.
[socket.io docs](https://socket.io/docs/v4/server-installation/#other-websocket-server-implementations)

**Comment by Ruby184** [comment, marked as answer]
This cannot be used with adonis because Adonis expects [node-style http server](https://nodejs.org/api/http.html#class-httpserver) to listen on. So it will not work because `req` and `res` are not compatible with node implementations.
You can boot and listen uWebsockets server inside adonis provider to bind socket.io on it, but it will have to use another port (different from adonis is listening on), so in my opinion there is no point of doing so.
